code/client/kstr.c

```diff
@@ -84,14 +84,15 @@ bool kstr_read_ascii(struct kstr *str, const char *src, size_t size)
 {
     str->length = 0;
     size_t length = strnlen(src, size);
-    if (str->capacity < length)
+    if (str->capacity < length + 1)
         return false;
-    for (size_t i = 0; i < length; i++) {
+    for (size_t i = 0; i < length && src[i]; i++) {
         if (src[i] & 0x80)
             return false;
         str->buffer[i] = src[i];
+        str->length++;
     }
-    str->length = length;
+    str->buffer[str->length++] = 0;

```

Comment:
This function is explicitly not null terminated to avoid relying on it, but also to permit null character inside. Why did you want to change that behavior?

Reply:
`kstr_read` indeed trim everything after `\0`, but this should include the `\0`, no? I will add some tests to set what we expect.

Regardless, I don't think we should have a `\0` in the buffer. Maybe you have a use case that I don't expect though.

Reply:
So client character names include the null character and this is done via kstr read, the problem is we read in the character we want from the command arg which uses kstr read ascii and doesn’t include it

Reply:
I added some tests and fixed two bugs. The commit is https://github.com/reduf/Headquarter/commit/b455fea22190be7715337967124beac3cf95de78 if you want to have a look.

It might have fixed your bug, can you check?